include/deal.II/multigrid/mg_tools.h

```diff
@@ -304,6 +331,72 @@ namespace MGTools
   double
   workload_imbalance(const Triangulation<dim, spacedim> &tria);
 
+  /**
+   * Similar to the above function but for a vector of triangulations,
+   * which might have been created by
+   * MGTransferGlobalCoarseningTools::create_geometric_coarsening_sequence().
+   */
+  template <int dim, int spacedim>
+  double
+  workload_imbalance(
+    const std::vector<std::shared_ptr<const Triangulation<dim, spacedim>>>
+      &trias);
+
+  /**
+   * Return the vertical communication cost between levels.
+   * The first returned vector contains for each level the number
+   * of cells that have the same owning process as their
+   * corresponding coarse cell (parent). The second one contains
+   * for each level the number of cells that have not the same
+   * owning process as their corresponding coarse cell.
+   */
+  template <int dim, int spacedim>
+  std::tuple<std::vector<types::global_dof_index>,
```

Comment:
Wouldn't a more natural data structure be `std::vector<std::pair<types::global_dof_index, types::global_dof_index>>`? I can't foresee a case where we would have a different number of vector entries, or widely different metrics that otherwise would merit a tuple/pair of vectors.